src/api.js

```diff
@@ -110,15 +125,23 @@ export default class Api {
       method: "POST",
       headers: DEFAULT_REQUEST_HEADERS,
       body: JSON.stringify({
-        defaults: { headers },
+        defaults: {headers: headers},
         requests: records.map(record => {
           const isDeletion = record._status === "deleted";
-          const path = this.endpoints({full: false}).record(collName, record.id);
+          const path = this.endpoints({full: false}).record(bucketName, collName, record.id);
           const method = isDeletion ? "DELETE" : "PUT";
-          const body = isDeletion ? undefined : cleanRecord(record);
-          const headers = options.safe ? {
-            "If-Unmodified-Since": String(record.last_modified || "0")
-          } : {};
+          const body = isDeletion ? undefined : { data: cleanRecord(record) };
+          const headers = {};
+          if (safe) {
+            if (record.last_modified) {
+              // Safe replace.
+              headers["If-Match"] = quote(record.last_modified);
+            }
+            else if (!isDeletion) {
```

Comment:
Nit: I think we usually prefered having the whole `else (if)` statement + brackets on a single line